quetz/utils.py

```diff
@@ -24,6 +30,74 @@ def add_entry_for_index(files, subdir, fname, data_bytes):
     )
 
 
+def parse_query(query):
+    accepted_filters = ['channel', 'description', 'summary']
+    query = unquote(query.strip())
+    regex = r"""(\S+:'(?:[^'\\]|\\.)*')|(\S+:\"(?:[^\"\\]|\\.)*\")|(-?\"(?:[^\"\\]|\\.)*\")|(-?'(?:[^'\\]|\\.)*')|\S+|\S+:\S+"""
```

Comment:
impressive! perhaps you might use named groups to make it more legible

also make sure that, you can skip any of the query terms. For example, you might search only by channel name:

```
channel=conda-forge,my-channel
```

it's possible that you already do this, i don't know

Reply:
yeah a bit more documentation on this regex would be great :)

also maybe we could write a small parser for this mini-language, that's not based on regex?

Reply:
Yes, it is not necessary to supply all the arguments. It's up to us how many we supply. I did give a demo of this today.
I partially borrowed this regex (and did some of my own modifications) so I will have to see how it works at a deeper level as well.

I can check docs about named groups. And, I need to investigate about making a parser a bit more.

Reply:
@madhur-tandon maybe using the Python tokenizer is even easier? It splits out "strings" with ticks already, and it's part of the standard library.

https://docs.python.org/3/library/tokenize.html

I am sure there are other tokenizer libraries implemented in Python around. That might be easier & more maintainable than this regex, in the long term :)

Reply:
We are now using `shlex` (https://docs.python.org/3/library/shlex.html) as suggested by @wolfv